### PR TITLE
Nav2 profile part 2

### PR DESCRIPTION
**Pull request by chrisnojima**
Note: This is a merge onto part 1 which mostly deals with pgp. I'm keeping it sep so the pr is slightly smaller

- [x] edit profile
- [x] add proof: username
- [x] add proof: website
- [x] add proof: post text
- [x] add proof: confirm/pending
- [x] test nav1
- [x] search
- [x] test mobile
- [x] test mobile nav1
- [x] view proof

--------
TODO part 3: these are in a different ticket now
- [ ] edit avatar
- [ ] add to team
- [ ] showcase team
- [ ] send
- [ ] chat
- [ ] mobile checklist

### CODE REVIEW

**Comment by chrisnojima** [comment]
split from https://github.com/keybase/client/pull/16506

**Comment by chrisnojima** [comment]
this pr actually fixes 2 issues w/ master
1. if you enter in a bad website name you can't continue on ever, its sorta dead
2. if you click to prove a website it takes you to the file flow and skips the file/dns interstitial page. this broke when we integrated the id3 suggestions

File: shared/actions/json/profile.json [hunk 1]
```diff
@@ -33,7 +33,6 @@
       "username": "string",
       "openWebsite?": "?boolean"
     },
-    "outputInstructionsActionLink": {},
```
Comment (chrisnojima) [inline shared/actions/json/profile.json]: unused

File: shared/actions/profile/index.js [hunk 2]
```diff
@@ -48,10 +48,10 @@ const finishRevoking = state => [
     assertion: state.config.username,
     guiID: TrackerConstants.generateGUIID(),
     inTracker: false,
-    reason: 'justRevoked',
+    reason: '',
```
Comment (chrisnojima) [inline shared/actions/profile/index.js]: don't actually put a string here, this is plumbed through and shows up

File: shared/actions/profile/index.js [hunk 3]
```diff
@@ -48,10 +48,10 @@ const finishRevoking = state => [
...
   }),
   ProfileGen.createRevokeFinish(),
-  RouteTreeGen.createNavigateUp(),
+  ...(flags.useNewRouter ? [] : [RouteTreeGen.createNavigateUp()]),
```
Comment (chrisnojima) [inline shared/actions/profile/index.js]: avoid nav in sagas now

File: shared/actions/profile/index.js [hunk 4]
```diff
@@ -107,41 +111,6 @@ const submitRevokeProof = (state, action) => {
   }
 }
 
-const openURLIfNotNull = (nullableThing, url, metaText) => {
```
Comment (chrisnojima) [inline shared/actions/profile/index.js]: doesnt' need to be in a saga

File: shared/actions/profile/proofs.js [hunk 5]
```diff
@@ -44,31 +45,28 @@ const recheckProof = (state, action) =>
     Tracker2Gen.createShowUser({asTracker: false, username: state.config.username})
   )
 
-const addProof = (_, action) => {
+function* addProof(_, action) {
```
Comment (chrisnojima) [inline shared/actions/profile/proofs.js]: before this was split into 2 functions that both handled the same event and used a switch to ensure only one ran, just make it one thing

File: shared/actions/profile/proofs.js [hunk 6]
```diff
@@ -100,26 +99,31 @@ function* addServiceProof(_, action) {
...
-        ProfileGen.createUpdateErrorText({
-          errorCode: null,
-          errorText: '',
-        })
-      )
-      const state = yield* Saga.selectState()
-      _promptUsernameResponse.result(state.profile.username)
-      _promptUsernameResponse = null
+    // loop since if we get errors we can get these events multiple times
+    while (true) {
```
Comment (chrisnojima) [inline shared/actions/profile/proofs.js]: the lack of this while loop causes this to not work in master. if you enter a bad web name and then try and continue it'll never keep moving forward as our take has died. this keeps it going while the rpc is still alive

File: shared/constants/tracker2.js [hunk 7]
```diff
@@ -132,18 +132,21 @@ export const rpcAssertionToAssertion = (row: RPCTypes.Identify3Row): Types.Asser
     value: row.value,
   })
 
-export const rpcSuggestionToAssertion = (s: RPCTypes.ProofSuggestion): Types.Assertion =>
-  makeAssertion({
-    assertionKey: s.key,
+export const rpcSuggestionToAssertion = (s: RPCTypes.ProofSuggestion): Types.Assertion => {
+  const ourKey = s.key === 'web' ? 'dnsOrGenericWebSite' : s.key
```
Comment (chrisnojima) [inline shared/constants/tracker2.js]: core tells us a suggestion of 'web' but internally we want to ask the user for a dns or web proof, with one row in the list, so we have this fake item. plumb this through so we get that screen back else dns proofs in the gui are impossible

File: shared/people/container.js [hunk 8]
```diff
@@ -22,7 +21,6 @@ const mapDispatchToPropsHeader = dispatch => ({
   onClickUser: (username: string) => dispatch(createShowUserProfile({username})),
   onSearch: () => {
     dispatch(createSearchSuggestions({searchKey: 'profileSearch'}))
-    dispatch(RouteTreeGen.createNavigateAppend({path: ['profileSearch']}))
```
Comment (chrisnojima) [inline shared/people/container.js]: search results are in an overlay and not a route now

File: shared/common-adapters/popup-dialog.desktop.js [hunk 9]
```diff
@@ -34,13 +34,15 @@ export function PopupDialog({
...
       >
         <Box style={Styles.collapseStyles([styles.container, fill && styles.containerFill, styleContainer])}>
-          <Icon
-            type="iconfont-close"
-            style={Styles.collapseStyles([styles.close, styleClose])}
-            color={Styles.globalColors.white}
-            onClick={onClose}
-            hoverColor={Styles.globalColors.white_40}
-          />
+          {onClose && (
```
Comment (chrisnojima) [inline shared/common-adapters/popup-dialog.desktop.js]: support not having a close button

File: shared/profile/confirm-or-pending/index.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
+// @flow
+import * as React from 'react'
```
Comment (chrisnojima) [inline shared/profile/confirm-or-pending/index.js]: unify desktop and mobile for all these screens 🔥

File: shared/profile/prove-enter-username/container.js [hunk 11]
```diff
@@ -1,31 +1,12 @@
...
 import * as ProfileGen from '../../actions/profile-gen'
-import React, {Component} from 'react'
+import * as RouteTreeGen from '../../actions/route-tree-gen'
 import ProveEnterUsername from '.'
 import {connect} from '../../util/container'
+import flags from '../../util/feature-flags'
 
 type OwnProps = {||}
 
-type State = {
```
Comment (chrisnojima) [inline shared/profile/prove-enter-username/container.js]: this state just moved into the wrapped component

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/profile/post-proof/index.js [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+    return (
+      <Modal onCancel={props.onCancel} skipButton={true}>
+        <Kb.ScrollView
+          style={styles.scroll}
+          contentContainerStyle={styles.scrollContent}
+          key={
+            props.errorMessage || 'scroll' /* if we get an error redraw entirely so we're not scrolled down */
+          }
+        >
+          {/* $FlowIssue  copyCapture not in box2 */}
```
Comment (buoyad) [inline shared/profile/post-proof/index.js]: doesn't `Box2` need to plumb this through to the `div`?

File: shared/profile/search/bar.js [hunk 13]
```diff
@@ -5,38 +5,43 @@ import * as Styles from '../../styles'
...
-          <Kb.Icon
-            color={Styles.globalColors.black_50}
-            fontSize={Styles.isMobile ? 20 : 16}
-            style={styles.searchIcon}
-            type="iconfont-search"
-          />
-          <Kb.Text style={styles.searchText} type="BodySemibold">
-            Search people
-          </Kb.Text>
+          <Kb.Box2 direction="horizontal">
```
Comment (buoyad) [inline shared/profile/search/bar.js]: `alignItems="center"`

File: shared/profile/search/bar.js [hunk 14]
```diff
@@ -5,38 +5,43 @@ import * as Styles from '../../styles'
...
+            <Kb.Icon
+              color={Styles.globalColors.black_50}
```
Comment (buoyad) [inline shared/profile/search/bar.js]: A little hard to read on profile headers
image
admin build:
image

File: shared/profile/search/bar.js [hunk 15]
```diff
@@ -5,38 +5,43 @@ import * as Styles from '../../styles'
...
+              fontSize={Styles.isMobile ? 20 : 16}
+              style={styles.searchIcon}
+              type="iconfont-search"
+            />
+            <Kb.Text style={styles.searchText} type="BodySemibold">
+              Search{Styles.isMobile ? '' : ' people'}
+            </Kb.Text>
+          </Kb.Box2>
         </Kb.ClickableBox>
         <Kb.Overlay
```
Comment (buoyad) [inline shared/profile/search/bar.js]: `FloatingBox` supports rendering into a `KeyboardAvoidingView`
```
  dest?: 'popup-root' | 'keyboard-avoiding-root',
```
but it's not plumbed through `Overlay`. The results list goes under the keyboard right now

File: shared/util/container.js [hunk 16]
```diff
@@ -13,6 +13,18 @@ export const networkErrorCodes = [
   constantsStatusCode.sctimeout,
 ]
 
+export const getRouteProps = (ownProps: any, key: string) => {
```
Comment (buoyad) [inline shared/util/container.js]: take master's

File: shared/profile/prove-website-choice/index.js [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+
+type Props = {|
+  onFile: () => void,
+  onDNS: () => void,
+  onCancel: () => void,
+|}
+
+const ProveWebsiteChoice = (p: Props) => (
+  <Modal onCancel={p.onCancel}>
+    <Kb.Box2 direction="vertical" fullWidth={true}>
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/profile/prove-website-choice/index.js]: these pages scroll and are super old so i don't really want to touch the contents too much. lets just leave them for now (not in this pr at least)

File: shared/profile/search/bar.js (hunk 14 above)
Comment (chrisnojima) [reply shared/profile/search/bar.js]: lets leave this for a bit. i'm sure we'll take a visual pass and adjust this stuff. currently this is used in the followers/following section and the proof section on profiles so it gets a whit eback ground and a color background so in order to fix that better we'd have to add more plumbing

File: shared/util/container.js (hunk 16 above)
Comment (chrisnojima) [reply shared/util/container.js]: done

**Pull request merged**